src/test/java/com/es/phoneshop/web/ProductListPageServletTest.java

```diff
@@ -35,7 +36,13 @@ public void setup(){
     @Test
     public void testDoGet() throws ServletException, IOException {
         servlet.doGet(request, response);
-
         verify(requestDispatcher).forward(request, response);
+        verify(request,times(1)).setAttribute(anyString(),any());
```

Comment:
Your servlet class should set "products" attribute equal to the list returned by dao. And the test should verify that